### DISCUSSION TITLE
Payment sheet is showing on IOS but on Android

### DISCUSSION THREAD

**Discussion by MuhammadSaad420** [flutter-stripe/flutter_stripe#982] [Q&A] [answered]
Description:
Payment sheet is working fine on IOS devices but not on Android devices. I have completed the android integration steps as describe in the [ReadMe](https://github.com/flutter-stripe/flutter_stripe/blob/main/README.md) file. Code is working fine.. i am able to initialize payment sheet. But i am not able to present payment sheet. Its also not showing any kind of exception
Following is the code
```
  Future<void> addPaymentMethod() async {
    String? customerId = locator<PaymentConfig>().getCustomerCred!.customerId;
    if (customerId != null) {
      Result<String, String?> result = await PayIn.createSetupIntent(customerId: customerId);
      result.when((errorCode) {
        PaymentExceptions.handleException(errorCode: errorCode);
      }, (secretKey) async {
        if (secretKey != null) {
        try {
          await Stripe.instance.initPaymentSheet(
              paymentSheetParameters: SetupPaymentSheetParameters(
                  setupIntentClientSecret: secretKey,
                  customerId: customerId,
                  style: ThemeMode.dark));
          await Stripe.instance.presentPaymentSheet();
          await getUserPaymentMethods();
          setBusy(true);
          await getUserPaymentMethods(save: true);
        } on Exception catch (_) {
          Loader.hideLoader();
        }
      }
      });
    }
  }
```

In the above code, when i am calling present payment sheet, it's not showing payment sheet nor its giving any error. After **await Stripe.instance.presentPaymentSheet()** no code is executing...

**Additional context**
Here in this code i am adding payment method using Setup Intent

**Comment by davidmigloz** [comment, marked as answer]
In my case I wasn't specifying any `merchantDisplayName`. After setting a name the PaymentSheet started working.
